container/src/main/java/org/wildfly/swarm/internal/SystemDependencyResolution.java

```diff
@@ -0,0 +1,144 @@
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.importer.ExplodedImporter;
+import org.jboss.shrinkwrap.api.importer.ZipImporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.wildfly.swarm.bootstrap.util.WildFlySwarmClasspathConf;
+
+/**
+ * @author Heiko Braun
+ * @since 18/07/16
+ */
+public class SystemDependencyResolution implements DependencyResolution {
+
+    public SystemDependencyResolution(List<String> classpath, String pwd, String javaHome, List<String> testClasspath) {
+        this.classpath = classpath;
+        this.pwd = pwd;
+        this.javaHome = javaHome;
+        this.testClasspath = testClasspath;
+    }
+
+    public SystemDependencyResolution setDryRun(boolean b) {
+        this.dryRun = b;
+        return this;
+    }
+
+    @Override
+    public Map<String, JavaArchive> resolve(List<String> exclusions) throws IOException {
+
+        final Map<String, JavaArchive> archives = new HashMap<>();
+
+        exclusions.replaceAll(s -> s.replace('.', File.separatorChar));
+
+        if (classpath != null) {
+            WildFlySwarmClasspathConf classpathConf = new WildFlySwarmClasspathConf();
+            Set<String> classpathElements = new HashSet<>();
+            Set<String> providedGAVs = new HashSet<>();
+            List<String> testClasspathElements = testClasspath!=null ?
+                    testClasspath : Collections.EMPTY_LIST;
+
+            // filter classpath entries, makes some assumptions about maven
+            // TODO: not sure how this actually works with gradle. For now I assume gradle works on the same same local repository ...
+
+            final String m2Repo = "m2" + File.separator + "repository";
+            final String projectDir = pwd + File.separatorChar;
+
+            for (final String element : classpath) {
+
+                // JAR files within CP scope
+                if(element.contains(m2Repo) && element.endsWith(JAR)) {
+                    // Read wildfly-swarm-classpath.conf entries
+                    if(!dryRun) {
+                        try (JarFile jar = new JarFile(new File(element))) {
+                            ZipEntry entry = jar.getEntry(WildFlySwarmClasspathConf.CLASSPATH_LOCATION);
+                            if (entry != null) {
+                                classpathConf.read(jar.getInputStream(entry));
+                            }
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                        }
+                    }
+
+                    if (!excluded(exclusions, element)) {
+                        //System.out.println("Adding " + element);
+                        classpathElements.add(element);
+                    }
+                }
+
+                // classes Dir's within scope (i.e. target/classes in case of maven build)
+                else if(element.startsWith(projectDir) && element.endsWith(CLASSES)) {
```

Comment:
With such an approach you will never catch all cases of directories with relevant classes in it - without interpreting the definition of the build tool. In gradle the build output folders have this pattern "build/[classes|resources]/<sourceset-name>" - for example "build/classes/main. 

At best you can try to evaluate the name of the corresponding in-memory-JARArchive for better "debugging" - that's the current logic in ArtifactManager which you moved to a default-method in DependencyResolution.